weather@mockturtl/files/weather@mockturtl/applet.js

```diff
@@ -409,9 +409,15 @@ MyApplet.prototype = {
         // Polling for likely API throttling
         Mainloop.timeout_add_seconds(5, Lang.bind(this, function() {
           this.refreshWeather(false)
-        }))  
+        }))
       }
+
       let weather = json.query.results.channel
+
+      if (!weather.item) {
+        return false
```

Comment:
I'm not familiar with JS idioms, but I would just log an error (warning?) and `return` here, or else change all `refreshWeather`'s callsites to consume a boolean result.